### PR TITLE
MODRTAC-17: Prevent parentheses with just whitespace in volume

### PR DESCRIPTION
**Pull request by mreno-EBSCO**
If an enumeration field is removed in the UI, instead of removing the value from the json, it sets the field to an empty string. Likewise, the enumeration fields allow white space only to be values, which will be stored in the json. The existing code treats these as usable values and will return the volume info as required, but as "()" or "(\<whitespaces\>)".

Now, we will not return volume information for whitespace only/empty string values, nor the parentheses that would surround the volume data.

Refactored the horrible `volume` building code to remove the misguided usage of the `Optional` class. Replaced the code with a more reasonable, easier to follow, if/else flow. Added a dependency on `commons-lang3` to handle the checks for empty and blank. Refactored some existing code to use the library as well. It will improve code coverage. 😄

### CODE REVIEW

File: src/test/java/org/folio/rest/impl/RtacResourceImplTest.java [hunk 1]
```diff
@@ -396,7 +396,7 @@ public final void testGetRtacByIdVolumeFormatting() {
     final JsonObject expectedJson = new JsonObject(
         readMockFile("RTACResourceImpl/success_rtac_response_volume_formatting.json"));
 
-    final int expectedSize = 7;
+    final int expectedSize = 8;
     assertEquals(expectedSize, json.getJsonArray("holdings").size());
     for (int i = 0; i < expectedSize; i++) {
       final JsonObject jo = json.getJsonArray("holdings").getJsonObject(i);
```
Comment (mtraneis) [inline src/test/java/org/folio/rest/impl/RtacResourceImplTest.java]: Not being picky but this "jo" object could be renamed to something more meaningful, such as "holdingsJo"

File: src/test/java/org/folio/rest/impl/RtacResourceImplTest.java [hunk 2]
```diff
@@ -396,7 +396,7 @@ public final void testGetRtacByIdVolumeFormatting() {
     final JsonObject expectedJson = new JsonObject(
```
Comment (mtraneis) [inline src/test/java/org/folio/rest/impl/RtacResourceImplTest.java]: This code wasn't touched, but if possible, the "json" object declared a line above (395) could be renamed to "respJson" or "responseJson" to make the name more meaningful.

File: src/main/java/org/folio/rest/impl/RtacResourceImpl.java [hunk 3]
```diff
@@ -250,17 +253,21 @@ private String getVolume(JsonObject item) {
...
-          .map(c -> sj.add(e).add(c).toString())
-          .orElse(sj.add(e).toString()))
-        .orElseGet(() -> volume
-            .map(v -> sj.add(v).toString())
-            .orElseGet(() -> chronology
-                .map(c -> sj.add(c).toString())
-                .orElse(null)));
+    final StringJoiner sj = new StringJoiner(" ", "(", ")").setEmptyValue("");
+
+    if (isNotBlank(enumeration)) {
```
Comment (mtraneis) [inline src/main/java/org/folio/rest/impl/RtacResourceImpl.java]: Thanks for rewriting this block so java novices like me can follow the code ;-)

File: src/test/java/org/folio/rest/impl/RtacResourceImplTest.java (hunk 1 above)
Comment (mreno-EBSCO) [reply src/test/java/org/folio/rest/impl/RtacResourceImplTest.java]: Do you think this clean up should be done with this story or with a tech debt story? I have no preference one way or the other... except maybe to minimize changes.

File: src/main/java/org/folio/rest/impl/RtacResourceImpl.java (hunk 3 above)
Comment (mreno-EBSCO) [reply src/main/java/org/folio/rest/impl/RtacResourceImpl.java]: Yeah, it was ridiculous as previously written. My bad...

File: src/test/java/org/folio/rest/impl/RtacResourceImplTest.java (hunk 1 above)
Comment (mtraneis) [reply src/test/java/org/folio/rest/impl/RtacResourceImplTest.java]: Let's create a tech debt story, but I have a feeling that we will not have time to work on tech debt stories. It took me a second look to follow the code correctly so it's not too bad, not 5 times :)

Comment (mreno-EBSCO) [reply src/test/java/org/folio/rest/impl/RtacResourceImplTest.java]: Done: [MODRTAC-27](https://issues.folio.org/browse/MODRTAC-27)

**Pull request merged**